[22.03] ath10k: backport encapsulation offload support

castiel652 opened the pull request:
This backports encap offload support from upstream.
On some ath10k devices there can be about 10% improvement on tx throughput.
Users can turn it on by setting frame_mode=2.

Signed-off-by: Zhijun You <hujy652@gmail.com>
(cherry picked from commit 00f64ed)

castiel652 commented:
@PolynomialDivision can you give it a test :)

johnnyslee commented:
I thought I only need the 4 new patches when I tested it on 22.03 previously. See diff here: https://github.com/johnnyslee/openwrt/compare/22.03-ath10k..castiel652:22.03
I can't figure out where those dropped lines came from.

castiel652 commented:
oops looks like I messed up when solving the conflict

Ansuel commented:
Mhhh don't know about backporting this...

ironpillow commented:
I tried it just now with ath10k mainline (no ct) and I am getting this:
`ath10k_pci: unknown parameter 'frame_mode' ignored`

I added it to:
`/etc/modules.d/ath10k/`

as
`ath10k_pci frame_mode=2`

Am I adding it in the right file?

ironpillow commented:
Is this correct:
```
/etc/modules.d# cat ath10k
ath10k_pci
options ath10k_core frame_mode=2
```

johnnyslee commented:
@ironpillow 
```
# cat /etc/modules.d/ath10k 
ath10k_core frame_mode=2
ath10k_pci
```
The above would be installed as part of `kmod-ath10k` if you cherry-pick 40c2cd8bd0f7e7b6553d47c11dacb48acc533863 as well

ironpillow commented:
Thank you! That works :)

ironpillow commented:
@johnnyslee 

Thank you, that works :)

castiel652 committed:
ath10k: backport encapsulation offload support

This backports encap offload support from upstream.
On some ath10k devices there can be about 10% improvement on tx throughput.
Users can turn it on by setting frame_mode=2.

Signed-off-by: Zhijun You <hujy652@gmail.com>
(cherry picked from commit 00f64ed)

castiel652 commented:
> Mhhh don't know about backporting this...

Can you explain why?

castiel652 commented:
> @castiel652 it's practically not widely tested. Don't know if it would be a good idea to backport it even if we use ath10k-ct by default

Fair point but how can we widely test it? ct is the default and I don't think casual users would know what's going on on master.

johnnyslee commented:
If ath10k-ct is the default (I'd never knew, wifi module and driver it uses are completely optional for my ClearFog Pro), then merging this won't affect normal users, right? Just state in release notes that we have this new optional feature for people who are interested to switch driver and test. Does this sound reasonable?

csharper2005 commented:
> it's practically not widely tested.

I notice that the speed on ath10k on Zyxel NBG6817 is 10-15% lower than on robimarko ath11k on AX3600.  In 802.11ac mode of course. -ct even worse.

So, let's do this wide test. I would like to get  my 10% improvement on tx throughput.

castiel652 commented:
> I notice that the speed on ath10k on Zyxel NBG6817 is 10-15% lower than on @robimarko's ath11k on AX3600. In 802.11ac mode of course. -ct even worse.

Why are you comparing it with ath11k?

csharper2005 commented:
> Why are you comparing it with ath11k?

Because I have it and can make test. Mt7621 with stock mediatek driver is also 10-15% faster that ath10 on 22.03-rc6.

ynezz commented:
> Mhhh don't know about backporting this...

I second your opinion. IMO it's too late, we're about to do the final release.
